### PR TITLE
Adjust formula for bin tick count

### PR DESCRIPTION
**Pull request by kanitw**
Merge https://github.com/vega/vega-lite/pull/4418 first.

This might be a better alternative to https://github.com/vega/vega-lite/pull/4394.

(I think we should later extract these `10` and `40` as constants.)

### CODE REVIEW

File: examples/compiled/interactive_concat_layer.vg.json [hunk 1]
```diff
@@ -255,7 +255,7 @@
           "grid": false,
           "title": "Rotten_Tomatoes_Rating (binned)",
           "labelOverlap": true,
-          "tickCount": {"signal": "ceil(concat_0_height/20)"},
+          "tickCount": {"signal": "ceil(concat_0_height/10)"},
```
Comment (domoritz) [inline examples/compiled/interactive_concat_layer.vg.json]: I don't think this is better

screen shot 2018-12-28 at 22 09 36

File: examples/compiled/layer_circle_independent_color.vg.json [hunk 2]
```diff
@@ -212,7 +212,7 @@
       "title": "Acceleration (binned)",
       "labelFlush": true,
       "labelOverlap": true,
-      "tickCount": {"signal": "ceil(width/20)"},
+      "tickCount": {"signal": "ceil(width/10)"},
```
Comment (domoritz) [inline examples/compiled/layer_circle_independent_color.vg.json]: I think this is just busier

screen shot 2018-12-28 at 22 10 29

**Comment by domoritz** [comment]
This is definitely the clean approach but I'm not yet convinced that every 10 pixels is better than every 20 pixels. Axes get quite busy after the change.

**Comment by kanitw** [comment]
It’s clearly better man. The new one facilitates readers to count number of
bins easily and see the bin boundary. Basically 20 is too high esp when
there are 12-14 bins.

**Comment by domoritz** [comment]
If your chart is larger, the 10px between ticks can be quite small. I feel it adds clutter. I can usually interpolate without problems even if I don't have e ticks every 10px.

**Comment by kanitw** [comment]
>  I can usually interpolate without problems even if I don't have e ticks every 10px.

You can, but it takes quite more effort to read it. Plus, my colleagues have mentioned that they get confused why we leave so much space between ticks instead of adding ticks between them. 

I think clutter is less important than having the ability the read the value right away.  I think it's too dense if we have more than 1 tick per 10 pixel though.

**Comment by kanitw** [comment]
@domoritz Any idea why Travis fails? I don't see any error in the log.

**Comment by kanitw** [comment]
@domoritz Any idea why Travis fails? I don't see any error in the log.

**Comment by domoritz** [comment]
Hmm, it's only one build: https://travis-ci.org/vega/vega-lite/jobs/473331462

**Comment by kanitw** [comment]
From conversation on slack, we agreed to merge this.

**Comment by domoritz** [comment]
To repeat the conversation. I still feel that 10px is tight but many users said they would like more ticks so we are changing this.

**Pull request merged**